### DISCUSSION TITLE
Dropping Support for Ubuntu 22.04, Qt 6.2, and FFmpeg 4.4

### DISCUSSION THREAD

**Discussion by RytoEX** [obsproject/obs-studio#11170] [Announcements]
# Overview

OBS Studio 31 will no longer support Ubuntu 22.04, Qt versions older than 6.4.2, and FFmpeg versions older than 6.1.1.

# Why Drop Ubuntu 22.04 and Qt 6.2?

As we continue to make changes to the UI code of OBS, it has become increasingly burdensome to support Qt 6.2. Currently, the only first-party distribution of OBS using Qt 6.2 is on Ubuntu 22.04.

Additionally, having to support older dependency versions such as FFmpeg 4.4 and nlohmann-json 3.10 on Ubuntu 22.04 has created similarly difficulties in maintaining our code.

This announcement amends our previous communication in #9055.

# When Will Things Happen?

## Timetable

| Timeframe         | Action                                                                                                      |
| ----------------- | ----------------------------------------------------------------------------------------------------------- |
| Starting Now      | Remove code for Qt < 6.4, FFmpeg < 6.1, other dependencies tied to Ubuntu 22.04, and Ubuntu 22.04 CI builds |
| Later             | Stable release of next major OBS version                                                                    |

## Starting Now

We will immediately begin work to:
1. Remove code in [obs-studio](https://github.com/obsproject/obs-studio) that enables building against Qt < 6.4, FFmpeg < 6.1, and other dependencies tied to Ubuntu 22.04.
2. Remove the Ubuntu 22.04 CI builds on [obs-studio](https://github.com/obsproject/obs-studio)

## Later

At a future point in time, we will release the next major version of OBS that will only support Qt 6.4+ and FFmpeg 6.1+.

# How Does This Affect Me?

## Users

Users of the first-party OBS distributions on Windows, macOS, Flatpak, or Ubuntu 24.04 will not see any impact. Ubuntu 22.04 users who wish to continue receiving new major/minor releases of OBS are encouraged to upgrade to Ubuntu 24.04 or later, or switch to using the Flatpak version of OBS.

## Package maintainers

We recognize that this change may impact third-party Linux distributions which currently use Qt 6.2 or 6.3. We encourage those to move to Qt 6.4+ along with us. If that is not possible, such as on LTS distributions, maintainers can stick to the `release/30.2` branch which will receive bug fixes at least until the release of the next major OBS version.

# Future considerations

We currently intend to support Qt 6.4 and Ubuntu 24.04 at least until Ubuntu 26.04 is released. This is subject to change, but we will provide another announcement like this if our plans should change.

**Comment by Kobi-Blade** [comment]
You should honestly be supporting the main reps of Linux, like Debian and Arch, not a fork like Ubuntu. Plus if it works on Debian, will work on Ubuntu.

Thankfully you only intent to drop QT 6.4 support with Ubuntu 26.04 release, by then Debian 13 will have shipped already with a newer QT version.

But I'm still concerned for a future where you'll neglect the main repos over unstable forks like Ubuntu.

**Comment by Blackstareye** [comment]
Thank you for the full transperency here.

Especially for this section
>Why Drop Ubuntu 22.04 and Qt 6.2?

Totally understandable for me.

**Comment by RytoEX** [comment]
These changes shipped with OBS Studio 31.0.0.
